From 4482874afb16ea9572250b537f1b1808fb50bc80 Mon Sep 17 00:00:00 2001
From: Haiyang Zhang <haiyangz@microsoft.com>
Date: Fri, 1 Sep 2017 14:30:06 -0700
Subject: [PATCH 92/98] hv_netvsc: Simplify the limit check in
 netvsc_set_channels()

Because of the following code, net->num_tx_queues equals to
VRSS_CHANNEL_MAX, and max_chn is less than or equals to VRSS_CHANNEL_MAX.

netvsc_drv.c:
alloc_etherdev_mq(sizeof(struct net_device_context),
                                VRSS_CHANNEL_MAX);
rndis_filter.c:
net_device->max_chn = min_t(u32, VRSS_CHANNEL_MAX, num_possible_rss_qs);

So this patch removes the unnecessary limit check before comparing
with "max_chn".

Signed-off-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/hyperv/netvsc_drv.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/net/hyperv/netvsc_drv.c	2017-10-05 14:20:56.167581873 -0700
+++ b/drivers/net/hyperv/netvsc_drv.c	2017-10-05 14:20:56.163581823 -0700
@@ -830,9 +830,6 @@ static int netvsc_set_channels(struct ne
 	    channels->rx_count || channels->tx_count || channels->other_count)
 		return -EINVAL;
 
-	if (count > net->num_tx_queues || count > VRSS_CHANNEL_MAX)
-		return -EINVAL;
-
 	if (!nvdev || nvdev->destroy)
 		return -ENODEV;
 
